### PR TITLE
Show owned noms

### PR DESCRIPTION
**Pull request by BriungRi**

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
 import { Moon, Sun } from "phosphor-react";
+import { AbiItem } from "@celo/connect";
+const Abi = require("src/abis/nomspace/Nom.json");
+
 
 export const Header: React.FC = () => {
   const location = useLocation();
   const [colorMode, setColorMode] = useColorMode();
+  const { address } = useContractKit();
+  const kit = newKit("https://forno.celo.org/%22");
```
Comment (BriungRi) [inline src/components/Header.tsx]: This should be `https://forno.celo.org`

You need to remove `/%22`

Comment (BriungRi) [reply src/components/Header.tsx]: Don't do this altogether.

Just do

```
const {kit, address} = useContractKit()
```
from line 16 and delete line 17

File: src/components/Header.tsx [hunk 2]
```diff
@@ -1,17 +1,41 @@
...
+        <Flex sx={{
+          alignItems: "center",
+          mr: 4,
+          px: 2,
+          cursor: "pointer",
+          borderRadius: "6px",
+        }}
+          onClick={async () => {
+            const events = await contract.getPastEvents("NameOwnerChanged", {
+              filter: { newOwner: "0xcf239C0c09e911d837f3b1349562dca2D0426669" }, // Using an array means OR: e.g. 20 or 23
```
Comment (BriungRi) [inline src/components/Header.tsx]: Replace `"0xcf239C0c09e911d837f3b1349562dca2D0426669"` with `address`

File: src/components/Header.tsx [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+              fromBlock: 0,
+              toBlock: 'latest',
+            });
```
Comment (BriungRi) [inline src/components/Header.tsx]: Returning items from an `onClick` does nothing. You need to display these items.

For now, just do `console.log(events)` to verify that you have indeed fetched the correct names.

File: src/components/Header.tsx (hunk 2 above)
Comment (Harshith246) [reply src/components/Header.tsx]: If I update it to address , getting the below error 

Overload 1 of 4, '(event: string, options: PastEventOptions): Promise<EventData[]>', gave the following error.
    Type 'string | null' is not assignable to type 'string | number | string[] | number[]'.
      Type 'null' is not assignable to type 'string | number | string[] | number[]'.

File: src/components/Header.tsx [hunk 4]
```diff
@@ -1,17 +1,46 @@
...
+              filter: { newOwner: address! }, // Using an array means OR: e.g. 20 or 23
+              fromBlock: 0,
+              toBlock: 'latest',
+            });
+            var arr: string[] = [];
+            events.forEach(event => {
+              const bytes = utils.parseBytes32String(event?.returnValues.name);
+              arr.push(bytes);
+            })
+            window.alert('your noms :' + arr);
```
Comment (BriungRi) [inline src/components/Header.tsx]: Fix the space. should be `'Your noms: ' + arr`

File: src/components/Header.tsx [hunk 5]
```diff
@@ -1,17 +1,46 @@
...
+        <Flex sx={{
+          alignItems: "center",
+          mr: 4,
+          px: 2,
+          cursor: "pointer",
+          borderRadius: "6px",
+        }}
+          onClick={async () => {
+            const events = await contract.getPastEvents("NameOwnerChanged", {
+              filter: { newOwner: address! }, // Using an array means OR: e.g. 20 or 23
```
Comment (BriungRi) [inline src/components/Header.tsx]: Don't use ! here. Instead, add a check for address before fetching:

```
if (!address) return;
const events = ...
```

File: src/components/Header.tsx [hunk 6]
```diff
@@ -1,17 +1,46 @@
...
+import { utils } from "ethers";
+const Abi = require("src/abis/nomspace/Nom.json");
+
+
 
 export const Header: React.FC = () => {
   const location = useLocation();
   const [colorMode, setColorMode] = useColorMode();
+  const { kit, address } = useContractKit();
+  const contract = new kit.web3.eth.Contract(Abi.abi as AbiItem[], "0xABf8faBbC071F320F222A526A2e1fBE26429344d");
```
Comment (BriungRi) [inline src/components/Header.tsx]: Import the Nom address from config or constants if possible

**Pull request merged**